site.cfg.example

```diff
@@ -52,6 +52,28 @@
 #       True) to tell numpy.distutils to prefer static libraries (.a) over
 #       shared libraries (.so). It is turned off by default.
 #           search_static_first = false
+#
+#   runtime_library_dirs/rpath
+#       List of directories that contains the libraries that should be 
+#       used at runtime, thereby disregarding the LD_LIBRARY_PATH variable.
+#       See 'library_dirs' for formatting on different platforms.
+#           runtime_library_dirs = /opt/blas/lib:/opt/lapack/lib
+#       or equivalently
+#           rpath = /opt/blas/lib:/opt/lapack/lib
+#
+#   extra_compile_args
+#       Add additional arguments to the compilation of sources.
+#       Simple variable with no parsing done. 
+#       Provide a single line with all complete flags.
+#           extra_compile_args = -g -ftree-vectorize
```

Comment:
My addition _only_ applies to codes/extensions using the `get_info` function from `numpy.distutils`. So I can't say if it is used for all numpy sources like `OPT`. See my comment on `lapack_lite` which shows one place where it would be used. So, these compile flags are _not_ used unless `get_info` is called on a section containing the above mentioned args.

I have added the `extra_compile_args` mainly for completeness sake, I see no disadvantage of having the possibility of adding extra flags when compiling extensions. You may say it is not necessary for libraries, but it allows people to construct sections which can be used to compiling extensions with different flags, say:

```
[O2]
extra_compile_args = -O2
[O3]
extra_compile_args = -O3
```

And then compiling can be done using:
```
config.add_extension('my_package',
                         sources = ['my_package.c'],
                         extra_info = get_info('O2')
                         )
config.add_extension('my_package3',
                         sources = ['my_package3.c'],
                         extra_info = get_info('O3')
                         )
```
One can of course extend several sections together.  
I like to be able to fully control my compilation of source when using Python, with this PR the user can add new sections to their `site.cfg` and then control all sources/extensions individually, if needed. 

Furthermore using config files, I think, is much more streamlining the build process than having to use env's. This is a matter of taste, but I see no disadvantage in letting users have a choice of this.